src/main/scala/com/databricks/spark/csv/package.scala

```diff
@@ -158,9 +158,13 @@ package object csv {
       // does not have to happen in the inner loop.
       val schema = dataFrame.schema
       val formatForIdx = schema.fieldNames.map(fname => schema(fname).dataType match {
-        case TimestampType => (timestamp: Any) =>
-          dateFormatter.format(new Date(timestamp.asInstanceOf[Timestamp].getTime))
-        case DateType => (date: Any) => dateFormatter.format(date)
+        case TimestampType => (timestamp: Any) => {
+          if (timestamp == null) nullValue
```

Comment:
Is this safe? It seems the type is expected as `Timestamp` but it is returning `String` here.